pilot/pkg/security/authz/model/generator.go

```diff
@@ -180,33 +181,33 @@ func (srcNamespaceGenerator) principal(_, value string, _ bool, useAuthenticated
 	return principalAuthenticated(m, useAuthenticated), nil
 }
 
-type srcServiceAccountGenerator struct{}
+type srcServiceAccountGenerator struct {
+	policyName types.NamespacedName
+}
 
-func (srcServiceAccountGenerator) permission(_, _ string, _ bool) (*rbacpb.Permission, error) {
+func (g srcServiceAccountGenerator) permission(_, _ string, _ bool) (*rbacpb.Permission, error) {
 	return nil, fmt.Errorf("unimplemented")
 }
 
-func (srcServiceAccountGenerator) principal(_, value string, _ bool, useAuthenticated bool) (*rbacpb.Principal, error) {
-	regex, err := serviceAccountRegex(value)
-	if err != nil {
-		return nil, err
-	}
+func (g srcServiceAccountGenerator) principal(_, value string, _ bool, useAuthenticated bool) (*rbacpb.Principal, error) {
+	regex := serviceAccountRegex(g.policyName.Namespace, value)
 	m := matcher.StringMatcherRegex(regex)
 	return principalAuthenticated(m, useAuthenticated), nil
 }
 
-func serviceAccountRegex(value string) (string, error) {
+func serviceAccountRegex(defaultNamespace string, value string) string {
 	ns, sa, ok := strings.Cut(value, "/")
 	if !ok {
```

Comment:
Maybe add a comment saying if the namespace isn't specified in the serviceAccount value, set it to the namespace of the policy